app/controllers/auth/registrations_controller.rb

```diff
@@ -22,13 +25,13 @@ def create
     build_resource(sign_up_params)
 
     resource.save
+    yield resource if block_given?
     if resource.persisted?
-      session_flash(resource, :signed_up)
-      sign_up(resource_name, resource)
-      respond_with resource, location: after_sign_up_path_for(resource), float: true
+      resource_persisted_response!
     else
-      redirect_to new_user_registration_path,
-                  alert: resource.errors.full_messages
+      clean_up_passwords resource
+      set_minimum_password_length
```

Comment:
Are those two methods above necessary? I didn't search enough to find a nice guide migration from devise 3 to 4, so...

Reply:
This is because the contents of this method is the exact code from Devise, plus some touches from us. The method changed quite a lot since then, so I simply updated the code by placing their code here, and applying our changes again.